store/src/main/java/com/dropbox/android/external/store4/impl/RealStore.kt

```diff
@@ -164,8 +164,13 @@ internal class RealStore<Key : Any, Input : Any, Output : Any>(
         val diskFlow = sourceOfTruth.reader(request.key, diskLock)
         // we use a merge implementation that gives the source of the flow so that we can decide
         // based on that.
-        return networkFlow.merge(diskFlow.withIndex())
-            .transform {
+        return networkFlow.merge(diskFlow.withIndex().onStart {
+            // wait for disk to latch first to ensure it happens before network triggers.
+            // after that, if we'll not read from disk, then allow network to continue
+            if (request.shouldSkipCache(CacheType.DISK)) {
+                networkLock.complete(Unit)
+            }
+        }).transform {
```

Comment:
Super easy to miss that the onStart in for the parameter and not the merge flow. Can you move it to line 164
```
val diskFlow = sourceOfTruth.reader(request.key, diskLock).onStart { ... }
```